brave/brave-ios
# Fix #4219: Add de-amp settings

## Summary of Changes


This pull request fixes #4219

## Submitter Checklist:


## Test Plan:
1. Disable de-amp setting
2. Go to google [search](https://www.google.com)
3. Do a search for "top stories"
4. In the top stories section some links will take you to a link similar to: `https://www.google.com/amp/s/{some_url}` (take note of the part in {some_url})
5. Go to the following links and you should not be redirected anywhere:
* `https://www.cbc.ca/amp/1.6432270`
* `https://fingerprint-protection-test.herokuapp.com/amp`
6. Enable de-amp settings
7. Go to the following urls and you should be redirected
* `https://www.cbc.ca/amp/1.6432270` -> `https://www.cbc.ca/news/{some-path}` (notice `/amp` is missing)
* `https://fingerprint-protection-test.herokuapp.com/amp` -> `https://amp.dev/documentation/guides-and-tutorials/start/create/basic_markup/`
8. Go to google [search](https://www.google.com)
9. Do a search for "top stories"
10. In the top stories section when you click on links you should **not** be sent to `https://www.google.com/amp/s/{some_url}`

## Screenshots:

How it looks like in desktop (for reference)
<img width="684" alt="Screen Shot 2022-04-21 at 12 23 28 PM" src="https://user-images.githubusercontent.com/909331/164569830-7e35e713-96a0-455f-85b9-4b089cd2e496.png">

This is how it looks in iOS:
![Simulator Screen Shot - iPad Air (5th generation) - 2022-04-27 at 15 09 21](https://user-images.githubusercontent.com/909331/165634078-9dbdd8c1-fbe2-4e87-96c3-8dfffce310db.png)

### Links
https://github.com/brave/security/issues/850

## Reviewer Checklist:

  - `QA/(Yes|No)`
  - `bug` / `enhancement`
